cheatsheets/gleam-for-scala-users.md

```diff
@@ -0,0 +1,1319 @@
+  - [Lists](#lists)
+  - [Maps](#maps)
+  - [Numbers](#numbers)
+- [Flow control](#flow-control)
+  - [Case](#case)
+  - [Piping](#piping)
+  - [Try](#try)
+- [Type aliases](#type-aliases)
+- [Custom types](#custom-types)
+  - [Records](#records)
+  - [Algebraic Data Types](#algebraic-data-types)
+  - [Opaque custom types](#opaque-custom-types)
+- [Type Parameters](#type-parameters)
+- [Modules](#modules)
+  - [Imports](#imports)
+  - [Named imports](#named-imports)
+  - [Unqualified imports](#unqualified-imports)
+- [Architecture](#architecture)
+
+## Comments
+
+### Scala
+
+In Scala, comments are written with a `//` prefix.
+
+```scala
+// Hello, Joe!
+```
+
+Multi line comments may be written like so:
+
+```scala
+/*
+ * Hello, Joe!
+ */
+```
+
+In Scala, above `trait`, `object`, `class`, `def` declarations
+there can be `scaladoc` like so:
+
+```scala
+/**
+ * a very special trait.
+ */
+trait Foo {}
+
+/**
+ * A Bar class
+ */
+class Bar {
+
+  /**
+   *
+   * @param s describes the s param
+   */
+  def quux(s: String) = {}
+
+}
+
+```
+
+Documentation blocks (scaladoc) are extracted into generated API
+documentation and use Markdown format.
+
+### Gleam
+
+In Gleam, comments are written with a `//` prefix.
+
+```gleam
+// Hello, Joe!
+```
+
+Comments starting with `///` are used to document the following function,
+constant, or type definition. Comments starting with `////` are used to
+document the current module.
+
+```gleam
+//// This module is very important.
+
+/// The answer to life, the universe, and everything.
+const answer: Int = 42
+
+/// A main function
+fn main() {}
+
+/// A Dog type
+type Dog {
+  Dog(name: String, cuteness: Int)
+}
+```
+
+`//` comments are not used while generating documentation files, while
+`////` and `///` will appear in them.
+
+## Variables
+
+Scala has mutable and immutable variable bindings; in Gleam, all data is immutable but
+variable names can be reassigned in the same scope.
+
+### Scala
+
+```scala
+val size = 50
+val size = size + 100 // compile error
+val size = 1 // compile error
+
+var size2 = 50
+size2 = size2 + 100
+size2 = 1
+```
+
+Scala uses `val` for immutable bindings and `var` for mutable bindings.
+Lexical scoping can be done with braces or indentation, and values with the same name
+can shadow each other.
+
+```scala
+
+def foo() {
+  val size = 50
+
+  def bar() {
+    val size = 100
+  }
+}
+
+```
+
+
+### Gleam
+
+Gleam has the `let` keyword before its variable names.
+
+```gleam
+let size = 50
+let size = size + 100
+let size = 1
+```
+
+### Match operator
+
+Both Scala and Gleam can use the left hand side of an `=` as a pattern.
+
+#### Scala
+
+```scala
+val List(a, b) = List(1, 2)
+// a == 1
+// b == 2
+```
+
+#### Gleam
+
+In Gleam, `let` and `=` can be used for pattern matching, but you'll get
+compile errors if there's a type mismatch, and a runtime error if there's
+a value mismatch. For assertions, the equivalent `let assert` keyword is
+preferred.
```

Comment:
`let assert` is required, not preferred. `let` does not result in a runtime error as it does not permit value mismatches

Reply:
I'll take a pass at re-writing this, but it's also included in the Elm, [Elixir](https://github.com/gleam-lang/website/blob/main/cheatsheets/gleam-for-elixir-users.md?plain=1#L106) and several other pages, so you might want to update those too.